### PR TITLE
Dev/readme

### PR DESCRIPTION
**Pull request by Luodian**
## Checklist

- [ ] Format your code
- [ ] Add unit tests
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,104 +1,395 @@
...
+
+</div>
+
+---
+
+## Overview
+
+**LMMs Engine** is a highly efficient, modular training framework for training Unified Multimodal Models at scale.
+
+Train any multimodal model architectures including vision-language models (Qwen2/3-VL, LLaVA-OV), diffusion models (dLLM, WanVideo), unified multimodal models (Qwen2.5-Omni, BAGEL) and specialized research architectures (RAE, Gated DeltaNet, SiT).
```
Comment (mwxely) [inline README.md]: I would suggest mentioning Qwen2.5/3-VL instead, as Qwen2-VL is somewhat outdated and no longer widely followed.

File: README.md [hunk 2]
```diff
@@ -1,104 +1,395 @@
...
+| **[RAE-SigLip](examples/representation_autoencoder)** | Visual AutoEncoder | ✅ | ❌ | ❌ | ❌ | ❌ | Representation AutoEncoder, LPIPS, EMA | [run.sh](examples/representation_autoencoder/run.sh) |
+| **[SiT](examples/scalable_interpolant_transformer)** | Diffusion Transformer | ✅ | ❌ | ❌ | ❌ | ❌ | Interpolant Transformer, CFG, ImageNet-1K | [run.sh](examples/scalable_interpolant_transformer/run.sh) |
+| **[WanVideo](examples/wanvideo)** | Video Generation | ✅ | ❌ | ❌ | ❌ | ❌ | T2V/I2V/V2V generation (1.3B/14B) | [run.sh](examples/wanvideo/run.sh) |
 
-### 1. Process the dataset
+**Optimization Legend:**
+- **FSDP2**: Fully Sharded Data Parallel v2 for distributed training
+- **Ulysses SP**: Sequence Parallel for long contexts
+- **Muon**: Advanced optimizer with Newton-Schulz orthogonalization
+- **Packing**: First-fit bin packing for peaking at 35-40% MFU vs 20-25% (wo in Qwen2.5-VL finetuning)
```
Comment (mwxely) [inline README.md]: wo → w/o

File: README.md [hunk 3]
```diff
@@ -1,104 +1,395 @@
...
-## Launch
+### Launch Training
 
-The recommended way to launch is always use torchrun as it is the most native way to launch torch and in most of the settings this should work. Most of the debug and development should be based on this as we might not always use accelerate in our later framework.
+**Recommended: torchrun (native PyTorch)**
+```bash
+torchrun --nproc_per_node=8 --nnodes=1 --node_rank=0 \
+  --master_addr=127.0.0.1 --master_port=12355 \
+  -m lmms_engine.launch.cli --config examples/qwen3_vl/example_config.yaml
+```
```
Comment (kcz358) [inline README.md]: The usage is wrong here. Should use config_yaml=xxx or use hydra --config-name and --config-path

File: README.md [hunk 4]
```diff
@@ -1,104 +1,395 @@
...
-torchrun --nproc_per_node=8 --nnodes=1 --node_rank=0 --master_addr=127.0.0.1 --master_port=12355 -m lmms_engine.launch.cli --config examples/load_from_pretrained_example.yaml
+accelerate launch --use_fsdp \
+  -m lmms_engine.launch.cli --config examples/qwen3_vl/example_config.yaml
 ```
 
-## Examples
+**Single GPU**
+```bash
+python -m lmms_engine.launch.cli --config examples/qwen3_vl/example_config.yaml
+```
```
Comment (kcz358) [inline README.md]: Same command args error

File: README.md [hunk 5]
```diff
@@ -1,104 +1,395 @@
...
+Production-grade efficiency from distributed training to kernel fusion.
+
+### Core Distributed Training
+
+- **FSDP2** - PyTorch 2.0+ DTensor-based sharding for parameters, gradients, and optimizer states. Improved composability over original FSDP enables flexible parallelism composition.
+
+- **Ulysses Sequence Parallel** - Splits sequence dimension across GPUs for ultra-long contexts. Critical for vision-language models like Qwen3-VL with 10K+ visual tokens.
+
+- **Multi-dimensional Parallelism** - Compose TP × CP × PP × DP meshes for cluster-scale training.
+
```
Comment (kcz358) [inline README.md]: This part should be listed as todos or not implemented

File: README.md [hunk 6]
```diff
@@ -1,104 +1,394 @@
...
+python -m lmms_engine.launch.cli config_yaml=examples/qwen3_vl/example_config.yaml
+```
 
-We provide two examples here to demonstrate how to use the training engine in most of the case, you will need to perform the following three steps:
+## 🔥 Featured Examples
 
-1. Process the dataset into a specific format and store it in (jsonl/json/arrow)
-2. Write your dataset yaml (Optional if you are only using a single data source)
-3. Prepare your training config
+| Model | Architecture | FSDP2 | Ulysses SP | Muon | Packing | NSA | Highlights | Quick Start |
```
Comment (BIGKnight) [inline README.md]: I think all models can be compatible with the muon optimizer, though they have not been fully tested yet.

File: README.md [hunk 7]
```diff
@@ -1,104 +1,394 @@
...
+| **[Gated DeltaNet](examples/dgn)** | Gated Linear Attn | ✅ | ❌ | ✅ | ✅ | ❌ | Efficient architecture, FineWeb-Edu pretraining | [run.sh](examples/dgn/run.sh) |
+| **[Qwen2.5-Omni](examples/qwen2_5_omni)** | Vision+Audio+Text | ✅ | ✅ | ✅ | ✅ | ❌ | Unified multimodal (image, audio, text) | [run.sh](examples/qwen2_5_omni/run.sh) |
+| **[Qwen3-VL](examples/qwen3_vl)** | Vision-Language | ✅ | ✅ | ✅ | ✅ | ❌ | Native-resolution, long context (10K+ tokens) | [run.sh](examples/qwen3_vl/run.sh) |
+| **[RAE-SigLip](examples/representation_autoencoder)** | Visual AutoEncoder | ✅ | ❌ | ❌ | ❌ | ❌ | Representation AutoEncoder, LPIPS, EMA | [run.sh](examples/representation_autoencoder/run.sh) |
+| **[SiT](examples/scalable_interpolant_transformer)** | Diffusion Transformer | ✅ | ❌ | ❌ | ❌ | ❌ | Interpolant Transformer, CFG, ImageNet-1K | [run.sh](examples/scalable_interpolant_transformer/run.sh) |
+| **[WanVideo](examples/wanvideo)** | Video Generation | ✅ | ❌ | ❌ | ❌ | ❌ | T2V/I2V/V2V generation (1.3B/14B) | [run.sh](examples/wanvideo/run.sh) |
 
-### 1. Process the dataset
+**Optimization Legend:**
+- **FSDP2**: Fully Sharded Data Parallel v2 for distributed training
```
Comment (BIGKnight) [inline README.md]: Do we also need to mention our support of Deepspeed?
